packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx

```diff
@@ -0,0 +1,229 @@
+import React, { useState } from 'react'
+import { cx } from '@linaria/core'
+import {
+  BodyText,
+  Button,
+  ButtonGroup,
+  ColSplit,
+  elM6,
+  elMb10,
+  elMt10,
+  FlexContainer,
+  FormLayout,
+  Grid,
+  Input,
+  InputAddOn,
+  InputGroup,
+  InputWrap,
+  InputWrapFull,
+  Label,
+  Modal,
+  PersistantNotification,
+  Subtitle,
+  Title,
+} from '@reapit/elements'
+import { IconContainer } from '../../webhooks/__styles__'
+import { WebhooksAnimatedNewIcon } from '../../webhooks/webhooks-animated-new-icon'
+import { WebhooksAnimatedDocsIcon } from '../../webhooks/webhooks-animated-docs-icon'
+import { ExternalPages, openNewPage } from '@/utils/navigation'
+import { mixed, object, string } from 'yup'
+import { AppTypeEnum, PackageManagerEnum, PipelineModelInterface } from '@reapit/foundations-ts-definitions'
+import { httpsUrlRegex, UpdateActionNames } from '@reapit/utils-common'
+import { useForm } from 'react-hook-form'
+import { yupResolver } from '@hookform/resolvers/yup'
+import errorMessages from '@/constants/error-messages'
+import { useReapitUpdate } from '@reapit/utils-react'
+import { reapitConnectBrowserSession } from '../../../../core/connect-session'
+import { useReapitConnect } from '@reapit/connect-session'
+
+export const pipelineCreateFormHandle =
+  (createPipeline: (values: Partial<PipelineModelInterface>) => Promise<boolean>, refresh: () => void) =>
+  async (values: PipelineModelInterface) => {
+    const result = await createPipeline({
+      ...values,
+      appType: AppTypeEnum.REACT,
+      name: values.repository,
+    })
+
+    if (result) refresh()
+  }
+
+const PipelineCreationModal = ({
+  open,
+  onModalClose,
+  appId,
+  refreshPipeline,
+}: {
+  open: boolean
+  onModalClose: () => void
+  appId: string
+  refreshPipeline: () => void
+}) => {
+  const { connectSession } = useReapitConnect(reapitConnectBrowserSession)
+  const schema = object().shape<PipelineModelInterface>({
+    repository: string()
+      .trim()
+      .required(errorMessages.FIELD_REQUIRED)
+      .matches(httpsUrlRegex, 'Should be a secure https url'),
+    buildCommand: string().trim().required('A build command is required'),
+    packageManager: mixed().oneOf(Object.values(PackageManagerEnum)),
+  })
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<PipelineModelInterface>({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      buildCommand: 'build',
+      packageManager: PackageManagerEnum.YARN,
+    },
+  })
+
+  const [loading, , send, submissionErrors] = useReapitUpdate<Partial<PipelineModelInterface>, PipelineModelInterface>({
```

Comment:
why the 2 commas? Are you not defining a var here?

Reply:
yea, eslint will hate me for not using it so didn't declare it 😂